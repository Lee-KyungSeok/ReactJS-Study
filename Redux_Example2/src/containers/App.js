import React, {Component} from 'react';

import Buttons from '../components/Buttons';
import CounterListContainer from './CounterListContainer';

import {connect} from 'react-redux';
import * as actions from '../actions';
import {getRandomColor} from "../util";


class App extends Component {
    render() {
        const {onCreate, onRemove} = this.props;
        return (
            <div className="App">
                <Buttons
                    onCreate={onCreate}
                    onRemove={onRemove}
                />
                <CounterListContainer/>
            </div>
        );
    }
}

const mapToDispatch = (dispatch) => ({
    onCreate: () => {
        const color = getRandomColor();
        dispatch(actions.create(color))
    },
    onRemove: (index) => dispatch(actions.remove(index))
});

export default connect(null, mapToDispatch)(App);