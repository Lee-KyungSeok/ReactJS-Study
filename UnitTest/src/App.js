import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import Counter from "./components/Counter";
import NameForm from "./components/NameForm";
import NameList from "./components/NameList";

class App extends Component {
    state = {
        names: ['Kyung', 'Seok']
    };

    onInsert = (name) => {
        const {names} = this.state;
        this.setState({
            names: [...names, name]
        })
    };

    render() {
        const {names} = this.state;
        const {onInsert} = this;
        return (
            <div>
                <Counter/>
                <NameForm onInsert={onInsert}/>
                <NameList names={names}/>
            </div>
        );
    }
}

export default App;
