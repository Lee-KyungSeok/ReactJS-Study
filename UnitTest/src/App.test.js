import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('NameList', () => {
    let component = null;

    it('renders correctly', () => {
        component = renderer.create(<App/>);
    });

    it('match correctly', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    //onInsert 확인
    it('insert correctly', () => {
        component.getInstance().onInsert('Test');
        expect(component.getInstance().state.names).toMatchObject(['Kyung','Seok','Test']);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});