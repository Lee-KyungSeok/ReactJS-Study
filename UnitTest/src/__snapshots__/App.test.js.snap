// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList insert correctly 1`] = `
<div>
  <div>
    <h1>
      카운터
    </h1>
    <h2>
      1
    </h2>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <label>
      이름
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
  <ul>
    <li>
      Kyung
    </li>
    <li>
      Seok
    </li>
    <li>
      Test
    </li>
  </ul>
</div>
`;

exports[`NameList match correctly 1`] = `
<div>
  <div>
    <h1>
      카운터
    </h1>
    <h2>
      1
    </h2>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <label>
      이름
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
  <ul>
    <li>
      Kyung
    </li>
    <li>
      Seok
    </li>
  </ul>
</div>
`;
